aura.grammar

```diff
@@ -0,0 +1,56 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl ',' decl_list | epsilon
+
+type: i[1-9][0-9]* | f16 | f32 | f64 | f128 | '[' type ']' | '{' type ':' type '}'
+
+
+array : '[' identifier_list ']'
+
+assignment_statement: identifier type '=' expression
+                    | '!' assignment_statement
+
+if_statement: 'if' expression NEWLINE body
+
+switch_statement: 'switch' identifier cases
```

Comment:
This should be the first official RFC, actually. The `next` keyword is in the same class as `break` and `continue`.

Switch cases in Arua have explicit fallthrough, which means unless you specify `next`, cases automatically break at the end of their execution body. `next` will jump to the next case.

There are two ways we could do this; have the `default` case, if any, enforced as the last case and not allow `next`. Or, have the `default` case allowed anywhere in the cases list and allow `next`. My hesitation with the latter approach is that the `default` case could be overlooked, but the former case doesn't allow for as flexible of switch/case structures.